feature_engine/encoding/decision_tree.py

```diff
@@ -261,16 +308,27 @@ def fit(self, X: pd.DataFrame, y: pd.Series):
         )
 
         # pipeline for the encoder
-        encoder_ = Pipeline(
+        pipe = Pipeline(
             [
-                ("categorical_encoder", cat_encoder),
-                ("tree_discretiser", tree_discretiser),
+                ("encoder", encoder),
+                ("tree", tree),
             ]
         )
 
-        encoder_.fit(X, y)
+        Xt = pipe.fit_transform(X, y)
+
+        encoder_ = {}
+        if self.precision is None:
+            for var in variables_:
+                encoder_[var] = dict(zip(X[var], Xt[var]))
```

Comment:
Just to be sure: the variable `variables_` doesn't contain duplicated values right?

Otherwise, this line of code could create some bugs. ;-)

Reply:
Great question, but It should not contain duplicates. 
`_variables` is derived from the [select_dtypes](https://github.com/feature-engine/feature_engine/blob/46155a0b51def2bf96c1c4719a3dd9bfd11f78d3/feature_engine/variable_handling/find_variables.py) function

Reply:
Thanks for the answer. 

It was just to be sure ;-)